src/uwuw/uwuw.cpp

```diff
@@ -0,0 +1,162 @@
+#include <unistd.h>
+#include <iostream>
+#include "uwuw.hpp"
+
+// Empty Constructor
+UWUW::UWUW() {
+  num_tallies = 0;
+  num_materials = 0;
+};
+
+// Default constructor
+UWUW::UWUW(char* file) {
+  std::string filename(file);
+
+  // turn the filename into a full filepath
+  full_filepath = get_full_filepath(filename);
+
+  if (!check_file_exists(full_filepath)) {
+    std::cerr << "The file " << full_filepath << " does not exist or is read protected" << std::endl;
+    exit(1);
+  }
+
+  // load materials
+  material_library = load_pyne_materials(full_filepath);
+
+  // load tallies
+  tally_library = load_pyne_tallies(full_filepath);
+};
+
+// Default constructor
+UWUW::UWUW(std::string filename) {
+  // turn the filename into a full filepath
+  full_filepath = get_full_filepath(filename);
+
+  // check for file existence
+  if (!check_file_exists(full_filepath)) {
+    std::cerr << "The file " << full_filepath << " does not exist or is read protected" << std::endl;
+    exit(1);
+  }
+
+  // load materials
+  material_library = load_pyne_materials(full_filepath);
+
+  // load tallies
+  tally_library = load_pyne_tallies(full_filepath);
+};
+
+// Destructor
+UWUW::~UWUW() {
+};
+
+// convert convert a filename into path+filename (for pyne)
+std::string UWUW::get_full_filepath(char* filename) {
+  std::string file(filename);
+  return UWUW::get_full_filepath(file);
+}
+
+// convert convert a filename into path+filename (for pyne)
+std::string UWUW::get_full_filepath(std::string filename) {
+  // remove all extra whitespace
+  filename.erase(std::remove(filename.begin(), filename.end(), ' '), filename.end());
+  // use stdlib call
+  const char* full_filepath = realpath(filename.c_str(), NULL);
+  return std::string(full_filepath);
+}
+
+// see if file exists
+bool UWUW::check_file_exists(std::string filename) {
```

Comment:
I've noticed many functions throughout this code base where an object instance, typically a `string` or `vector`, is passed by value when it could be passed by reference, especially const reference. eg. this function could be changed to have the following signature for greater performance, optimization potential, and guarantee against modification of the passed-in string:
```c++
bool UWUW::check_file_exists(const std::string& filename);
```